staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -208,6 +208,8 @@ func (f *fancyResponseWriterDelegator) Flush() {
 }
 
 func (f *fancyResponseWriterDelegator) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+	// fake a response status before protocol switch happens
+	f.processCode(http.StatusSwitchingProtocols)
```

Comment:
Which means that in the case of Hijacking we actually have no clue what will happen. The hijacker might write headers or another code, i.e. manually via "binary" write. The only thing we can do is to introduce another audit stage without any http code (similar to the panic stage). /cc @timstclair @soltysh @CaoShuFeng is it worth it to add that?